test/Test/Database/LSMTree/Model/Table.hs

```diff
@@ -0,0 +1,99 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+
+module Test.Database.LSMTree.Model.Table (tests) where
+
+import qualified Data.ByteString as BS
+import qualified Data.Vector as V
+import           Database.LSMTree.Common
+import           Database.LSMTree.Model.Table (LookupResult (..), Table,
+                     Update (..), lookups, retrieveBlobs)
+import qualified Database.LSMTree.Model.Table as Model
+import           Database.LSMTree.Monoidal (ResolveValue (..),
+                     resolveDeserialised)
+import           GHC.Exts (IsList (..))
+import           Test.QuickCheck.Instances ()
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+tests :: TestTree
+tests = testGroup "Database.LSMTree.Model.Table"
+    [ testProperty "lookup-insert" prop_lookupInsert
+    , testProperty "lookup-delete" prop_lookupDelete
+    , testProperty "insert-insert" prop_insertInsert
+    , testProperty "upsert-insert" prop_upsertInsert
+    , testProperty "upsert=lookup+insert" prop_upsertDef
+    , testProperty "insert-commutes" prop_insertCommutes
+    ]
+
+type Key = BS.ByteString
+
+newtype Value = Value BS.ByteString
+  deriving stock (Eq, Show)
+  deriving newtype (Arbitrary, SerialiseValue)
+
+instance ResolveValue Value where
+    resolveValue = resolveDeserialised resolve
+
+resolve :: Value -> Value -> Value
+resolve (Value x) (Value y) = Value (x <> y)
+
+type Blob = BS.ByteString
+
+type Tbl = Table Key Value Blob
+
+updates :: V.Vector (Key, Update Value Blob) -> Table Key Value Blob -> Table Key Value Blob
+updates = Model.updates Model.getResolve
+
+inserts :: V.Vector (Key, Value, Maybe Blob) -> Table Key Value Blob -> Table Key Value Blob
+inserts = Model.inserts Model.getResolve
+
+deletes :: V.Vector Key -> Table Key Value Blob -> Table Key Value Blob
+deletes = Model.deletes Model.getResolve
+
+mupserts :: V.Vector (Key, Value) -> Table Key Value Blob -> Table Key Value Blob
+mupserts = Model.mupserts Model.getResolve
```

Comment:
Without the `ResolveValue -> ResolveSerialisedValue` change, we would not need these functions

Reply:
Yeah, it adds a little bit of boilerplate here (didn't want to make the properties themselves more noisy), but I think it's overall still nicer than having spurious `ResolveValue` constraints show up.